smt.go

```diff
@@ -91,6 +96,54 @@ func (smt *SparseMerkleTree) Get(key []byte) ([]byte, error) {
 	return value, nil
 }
 
+// GetLeaf gets an entire leaf node from the tree.
+func (smt *SparseMerkleTree) GetLeaf(key []byte) (*LeafNode, error) {
+	if bytes.Equal(smt.root, smt.th.placeholder()) {
+		// The tree is empty, return the default value.
+		return nil, nil
+	}
+
+	path := smt.th.path(key)
+	currentHash := smt.root
+	for i := 0; i < smt.depth(); i++ {
```

Comment:
This is outside the scope of this PR butI wonder if it would be worthwhile to create a secondary index, internal to the SMT, that maps a key/path directly to the associated leaf node so that we don't have to iterate down a branch from the root- performing multiple db lookups- to  get to the leaf (or discover it is nonexistent).